lb.tf

```diff
@@ -39,9 +80,92 @@ data "exoscale_compute_template" "ubuntu2004" {
   name = "Linux Ubuntu 20.04 LTS 64-bit"
 }
 
+resource "null_resource" "register_lb" {
+  triggers = {
+    # Refresh resource when script changes -- this is probaby not required for production
+    script_sha1 = filesha1("${path.module}/files/register-server.sh")
+    # Refresh resource when lb fqdn changes
+    lb_id = local.instance_fqdns[count.index]
+  }
+
+  count = var.lb_count
+
+  provisioner "local-exec" {
+    command = "${path.module}/files/register-server.sh"
+    environment = {
+      CONTROL_VSHN_NET_TOKEN = var.control_vshn_net_token
+      SERVER_FQDN            = local.instance_fqdns[count.index]
+      # This assumes that the first part of var.region is the encdata region
+      # (country code for Exoscale).
+      SERVER_REGION = split("-", var.region)[0]
+      # The encdata service doesn't allow dashes, so we replace them with
+      # underscores.
+      # This assumes that any zone configurations already exist in Puppet
+      # hieradata.
+      SERVER_ZONE = replace(var.region, "-", "_")
+      # Cluster id is used as encdata stage
+      CLUSTER_ID = var.cluster_id
+    }
+  }
+}
+
+resource "gitfile_checkout" "appuio_hieradata" {
+  repo = "https://${var.hieradata_repo_user}@git.vshn.net/appuio/appuio_hieradata.git"
```

Comment:
Is it intentional that this terraform module would then only be able to install LBs for the APPUiO tenant?
For other tenants/customers, we wouldn't be able to install loadbalancers?

Reply:
As far as I understand, the OCP4 managed LBs will always be provisioned in the APPUiO tenant, even when the cluster belongs to another tenant.

Reply:
Yes, that's the idea, as the LBs are a black-box for tenant of the cluster.